### PR TITLE
support python3.12 now, so use bbhx master

### PR DESCRIPTION
**Pull request by WuShichao**
## Standard information about the request

This is a: plugin update

This change affects: inference, LISA-related

This change changes: documentation, result presentation / plotting, scientific output

This change will: almost no changes

## Motivation

Supports the latest BBHx version.

- [x] The author of this pull request confirms they will adhere to the [code of conduct](https://github.com/gwastro/pycbc/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -26,7 +26,7 @@ deps =
     {[base]deps}
     pytest
     ; Needed for `BBHx` package to work with PyCBC
-    git+https://github.com/mikekatz04/BBHx.git@4fff509; sys_platform == 'linux'
+    git+https://github.com/mikekatz04/BBHx.git; sys_platform == 'linux'
```
Comment (GarethCabournDavies) [inline tox.ini:29]: This change needed on line 70 as well

Comment (WuShichao) [reply tox.ini:29]: OK, also remove it.

**Comment by GarethCabournDavies** [comment]
obviously there have been a lot of changes to CI on master since this

so this can be either closed so it can be developed, or repurposed for experimenting with setting up the BBHx in CI again

**Comment by titodalcanton** [comment]
On a quick inspection of BBHx, I am not convinced that it functionally requires Python 3.12. With a minor edit I managed to install it on a 3.9 virtualenv, though I have not tested it. My change is on this branch:

https://github.com/titodalcanton/BBHx/tree/py39-and-cleanup

@WuShichao could you try using that branch instead of master here, re-enabling the LISA tests that were disabled in #4833, and see if they pass?

**Comment by WuShichao** [comment]
Reopen with Tito's BBHx.

File: tox.ini [hunk 2]
```diff
@@ -26,7 +26,7 @@ deps =
     {[base]deps}
     pytest
     ; Needed for `BBHx` package to work with PyCBC
-    ; git+https://github.com/mikekatz04/BBHx.git@4fff509; sys_platform == 'linux'
+    ; git+https://github.com/titodalcanton/BBHx/tree/py39-and-cleanup; sys_platform == 'linux'
```
Comment (titodalcanton) [inline tox.ini]: I think this needs to be
```suggestion
    git+https://github.com/titodalcanton/BBHx.git@py39-and-cleanup; sys_platform == 'linux'
```

File: tox.ini [hunk 3]
```diff
@@ -26,8 +26,8 @@ deps =
     {[base]deps}
     pytest
     ; Needed for `BBHx` package to work with PyCBC
-    ; git+https://github.com/mikekatz04/BBHx.git@4fff509; sys_platform == 'linux'
-    ; git+https://github.com/gwastro/BBHX-waveform-model.git; sys_platform == 'linux'
+    git+https://github.com/titodalcanton/BBHx/tree/py39-and-cleanup; sys_platform == 'linux'
```
Comment (titodalcanton) [inline tox.ini:29]: I think this needs to be
```suggestion
    git+https://github.com/titodalcanton/BBHx.git@py39-and-cleanup; sys_platform == 'linux'
```

Comment (WuShichao) [reply tox.ini]: Let me reable those LISA examples in doc.

**Comment by titodalcanton** [comment]
Do not forget to reenable the tests, i.e. basically revert #4833 apart from the change to `setup.py`.

**Comment by WuShichao** [comment]
> Ok, progress… @GarethCabournDavies @WuShichao I will let you decide if you want to merge this as-is now, or get BBHx to correct its Python requirements upstream.

The current solution seems work, we can merge this for CI tests and doc, fix from upstream will take some time.

**Review by titodalcanton — APPROVED** [review]

**Comment by GarethCabournDavies** [comment]
Just getting back to it, thanks both of you for sorting this!

**Pull request merged**